### ISSUE TITLE
Docker won't start after Mac OSX upgrade

### ISSUE THREAD

**Issue by ofostier** [docker/for-mac#630]
### Information

Diagnostic ID: 71B65A8E-E8D9-428D-B8C6-2D6B528FBD58
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.6 (Build 15G1004)
[ERROR]  docker-cli
         docker ps failed
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux
### Steps to reproduce
1. ... Update to Mac OXS 10.11.6
2. ... Reboot
3. ... Try to start Docker

**Comment by samoht** [comment]
Detected symptom of problem 'invariant-violated' in 71B65A8E-E8D9-428D-B8C6-2D6B528FBD58/20160915-103454.

An invariant within the code of one of the components was not held. Docker needs to be restarted.

May be related to docker/for-mac#89

The following log matches:
```
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:26 +0000 UTC daemon.info vsudd: 410 Done. read: 289 written: 925216
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.osxfs][44049] <Notice>: Volume.stop a3db2b689ab0cd4cf6ca8213ebe3512cbc3151b46544b24d3d8c988db400cd7b (paths = [/Users/131562/PhpstormProjects/docker-elk-load/kibana/config])
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.osxfs][44049] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:26 +0000 UTC daemon.info vsudd: Connection 425 to: 2376 from: 00000000
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:26 +0000 UTC daemon.info vsudd: Connection 426 to: 2376 from: 00000000
Sep 12 15:52:27 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:27 +0000 UTC kern.info kernel: docker0: port 2(vethac85a39) entered disabled state
``````
Detected symptom of problem 'invariant-violated' in 71B65A8E-E8D9-428D-B8C6-2D6B528FBD58/20160915-103454.

An invariant within the code of one of the components was not held. Docker needs to be restarted.

May be related to docker/for-mac#89

The following log matches:
```
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:26 +0000 UTC daemon.info vsudd: 410 Done. read: 289 written: 925216
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.osxfs][44049] <Notice>: Volume.stop a3db2b689ab0cd4cf6ca8213ebe3512cbc3151b46544b24d3d8c988db400cd7b (paths = [/Users/131562/PhpstormProjects/docker-elk-load/kibana/config])
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.osxfs][44049] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:26 +0000 UTC daemon.info vsudd: Connection 425 to: 2376 from: 00000000
Sep 12 15:52:26 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:26 +0000 UTC daemon.info vsudd: Connection 426 to: 2376 from: 00000000
Sep 12 15:52:27 twonkynmcstationarymbpdeolrfostier Docker[com.docker.driver.amd64-linux][44051] <Notice>: VM: 2016-09-12 13:52:27 +0000 UTC kern.info kernel: docker0: port 2(vethac85a39) entered disabled state
```

@dsheets is this fixed in 1.12.1?

**Comment by dsheets** [comment]
This invariant violation was a symptom of a use-after-free bug in the file system event watching code which occurred due to multiple containers with volume mounts racing to set up their mounts. It is possible that it caused hangs or crashes and has been fixed in the recently released Beta 29.3.

It is possible that this defect was the root cause of your issue but if no containers were run after the system upgrade and the system upgrade was the precipitating event then it is unlikely that the mount race is the cause.

@ofostier do you think that the mount race could be responsible or was it really the OS upgrade that immediately preceded the fault? I'm marking this as 0-more-info-needed so this issue will automatically close in 2 weeks if we receive no information from you.

**Comment by ijc25** [comment]
This issue has been inactive for more than 14 days while marked as `status/0-more-info-needed`. It is being closed due to abandonment. Please feel free to re-open with more information about the problem. 

MORE_INFO_EXPIRY_TIMEOUT
